library/Icingadb/Hook/Common/TotalSlaReportUtils.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/* Icinga DB Web | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Hook\Common;
+
+use Icinga\Module\Icingadb\ProvidedHook\Reporting\HostSlaReport;
+use Icinga\Module\Icingadb\Widget\EmptyState;
+use Icinga\Module\Reporting\Timerange;
+use ipl\Html\Html;
+
+use function ipl\I18n\t;
+
+trait TotalSlaReportUtils
+{
+    public function getHtml(Timerange $timerange, array $config = null)
+    {
+        $data = $this->getData($timerange, $config);
+
+        if (! count($data)) {
+            return new EmptyState(t('No data found.'));
+        }
+
+        $threshold = isset($config['threshold']) ? (float) $config['threshold'] : static::DEFAULT_THRESHOLD;
```

Comment:
You could also use this:

```suggestion
        $threshold = (float) ($config['threshold'] ?? static::DEFAULT_THRESHOLD);
```

The parentheses around the null coalescing operation are required as otherwise the float cast is applied only to `$config['threshold']` resulting in the cast from `NULL` to `0.0`.